### PR TITLE
Don't checkAlias abstract definitions

### LINKED ISSUE #418: Unifier ignores presence of abstract keyword
``` agda
module Bug where

data _≡_ (A : Set₁) : Set₁ → Set₂ where
  refl : A ≡ A

abstract

  A : Set₁
  A = Set

  unfold-A : A ≡ _
  unfold-A = refl

-- I don't think we should solve the meta-variable corresponding to
-- the underscore above. We have two obvious choices, A and Set, and
-- these choices are not equivalent.

```

Original issue reported on code.google.com by `nils.anders.danielsson` on 25 May 2011 at 3:30

### PR DESCRIPTION
**Pull request by UlfNorell**
Fixes #418 even more

### CODE REVIEW

**Review by andreasabel — CHANGES_REQUESTED** [review]
- The user manual needs update.  Currently, it uses many abstract aliases.  It would be good to add an explanation that (and why) this is no longer supported.
- Please add a changelog entry.
- The implementation now simply pretends not to recognize an alias in an abstract block.  For better UX and migration to 2.6.2, I think a warning should be emitted when aliases are put in abstract blocks (they will inevitably lead to unsolved metas).

File: test/Fail/Tests.hs [hunk 1]
```diff
@@ -61,7 +61,7 @@ mkFailTest agdaFile =
   updGolden  = writeTextFile goldenFile
 
   doRun = do
-    let agdaArgs = ["-v0", "-i" ++ testDir, "-itest/" , agdaFile
+    let agdaArgs = ["-v0", "-vwarning:1", "-i" ++ testDir, "-itest/" , agdaFile
```
Comment (andreasabel) [inline test/Fail/Tests.hs:64]: Ah, this explains the additional warnings unrelated to `abstract`.

**Review by andreasabel — COMMENTED** [review]
Thanks!
Somehow no checks are reported on this PR.  Weird, shouldn't CI kick in here?
I would think that some abstract aliases in the user manual need now type signatures.  The first one I spot is: https://github.com/agda/agda/blob/cb1ea6808e5a458b16a50772ab8b17435ec04c19/doc/user-manual/language/abstract-definitions.lagda.rst#L65

**Comment by UlfNorell** [comment]
> I would think that some abstract aliases in the user manual need now type signatures.

Those were all fixed in the original commit.

**Comment by UlfNorell** [comment]
Not sure what's up with CI. If you go to the Checks tab it says "Starting your workflow runs"

File: test/Fail/Tests.hs (hunk 1 above)
Comment (UlfNorell) [reply test/Fail/Tests.hs:64]: I was mystified for a little bit as to why my new warnings didn't show up in the golden value.

**Comment by UlfNorell** [comment]
I'll force push a new hash for the last commit and see if that triggers a run.

**Comment by UlfNorell** [comment]
That worked. It seems like github actions are a little flaky at the moment. On the other PR, I had one check fail with "This check failed", and a big X and no further information.

**Comment by andreasabel** [comment]
> > I would think that some abstract aliases in the user manual need now type signatures.
> 
> Those were all fixed in the original commit.

Ah, sorry for the noise then.

Thanks!

**Review by andreasabel — APPROVED** [review]

**Pull request merged**